cmd/erasure-object.go

```diff
@@ -1255,19 +1255,33 @@ func (er erasureObjects) DeleteObjects(ctx context.Context, bucket string, objec
 				}
 				return
 			}
-			errs := disk.DeleteVersions(ctx, bucket, dedupVersions)
-			for i, err := range errs {
-				if err == nil {
+			dresp := disk.DeleteVersions(ctx, bucket, dedupVersions)
+			for i, resp := range dresp {
+				if resp.NoOp {
+					// overlay delete marker version ID with top most delete marker on the stack
+					// for the version
+					for _, v := range dedupVersions[i].Versions {
+						if !v.Deleted {
+							continue
+						}
+						oi, err := er.getObjectInfo(ctx, bucket, dobjects[v.Idx].ObjectName, ObjectOptions{})
+						if err != nil && oi.DeleteMarker {
+							dobjects[v.Idx].DeleteMarkerVersionID = oi.VersionID
+							continue
+						}
+					}
+				}
+				if resp.Err == nil { // check if all err are nil
```

Comment:
We should return the top-most FileInfo as part of this call in DeleteVersions() and calculate quorum on it instead of calling getObjectInfo again..